src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStoreV1.swift

```diff
@@ -854,26 +888,74 @@ class ContactDiaryStoreV1: DiaryStoring, DiaryProviding {
 	}
 }
 
+// MARK: Creation
+
 extension ContactDiaryStoreV1 {
-	convenience init(fileName: String) {
+
+	static func make(filename: String) -> ContactDiaryStoreV1 {
+		Log.info("[ContactDiaryStore] Trying to create contact diary store...", log: .localData)
+
+		if let store = ContactDiaryStoreV1(fileName: filename) {
+			Log.info("[ContactDiaryStore] Successfully created contact diary store", log: .localData)
+			return store
+		}
+
+		Log.info("[ContactDiaryStore] Failed to create contact diary store. Try to rescue it...", log: .localData)
+
+		// The database could not be created – To the rescue!
+		// Remove the database file and try to init the store a second time.
+		try? FileManager.default.removeItem(at: ContactDiaryStoreV1.storeDirectoryURL)
+
+		if let secondTryStore = ContactDiaryStoreV1(fileName: filename) {
+			Log.info("[ContactDiaryStore] Successfully rescued contact diary store", log: .localData)
+			return secondTryStore
+		} else {
+			Log.info("[ContactDiaryStore] Failed to rescue contact diary store.", log: .localData)
+			fatalError("[ContactDiaryStore] Could not create contact diary store after second try.")
+		}
+	}
+
+	convenience init?(fileName: String) {
```

Comment:
This one is only used from the make method as well - still only internal?

Reply:
Its still a possible, convenience way to create the store. One could use it in testing to avoid the rescue logic in make.